### PR TITLE
Add M4cs Keymap for dekunukem/duckyPad

### PR DESCRIPTION
**Pull request by M4cs**
This is a pull request containing my keymap for the duckyPad by @dekunukem. Includes 3 layers w/ RGB settings as well.

## Description

- Adds keymap for duckypad
- Adds dekunukem/duckypad folder

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dekunukem/duckypad/config.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "config_common.h"
+
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0B91
+#define DEVICE_VER      0x0001
+#define MANUFACTURER dekuNukem
+#define PRODUCT duckyPad
+#define DESCRIPTION Open Source Keypad
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/config.h:10]:
```suggestion
```

File: keyboards/dekunukem/duckypad/config.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dekunukem/duckypad/duckypad.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#include "duckypad.h"
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/duckypad.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dekunukem/duckypad/duckypad.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+#include "stdio.h"
+
+#ifdef RGB_MATRIX_ENABLE
+#include "rgb_matrix.h"
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/duckypad.c:6]: This shouldn't be needed, if quantum.h is included in the keyboard's header file, properly. 
```suggestion
```

File: keyboards/dekunukem/duckypad/duckypad.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#include "duckypad.h"
+
+#include "stdio.h"
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/duckypad.c:3]: Not sure that this is needed. 

```suggestion
```

File: keyboards/dekunukem/duckypad/duckypad.c [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+#endif
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A0);
+    writePinHigh(A0);
+    writePinLow(A0);
+    wait_ms(10);
+    writePinHigh(A0);
+    rgb_matrix_toggle_noeeprom();
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/duckypad.c:38]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
    rgb_matrix_toggle_noeeprom();
#endif
```

File: keyboards/dekunukem/duckypad/duckypad.h [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/duckypad.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dekunukem/duckypad/keymaps/m4cs/config.h [hunk 8]
```diff
@@ -0,0 +1,62 @@
+#ifndef NO_DEBUG
+#define NO_DEBUG
+#endif // !NO_DEBUG
+#if !defined(NO_PRINT) && !defined(CONSOLE_ENABLE)
+#define NO_PRINT
+#endif // !NO_PRINT
+
+#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/config.h:8]:
```suggestion
#pragma once
```

The rest shouldn't be needed if console isn't enabled. 
And needs a license header.

File: keyboards/dekunukem/duckypad/keymaps/m4cs/rules.mk [hunk 9]
```diff
@@ -0,0 +1,4 @@
+LTO_ENABLE = yes
+
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/rules.mk:3]: The above LTO_ENABLE adds this. Not needed here.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by M4cs** [comment]
Made changes based on your comments. All have been resolved.

**Review by drashna — APPROVED** [review]

File: keyboards/dekunukem/duckypad/info.json [hunk 10]
```diff
@@ -0,0 +1,31 @@
+{
+    "keyboard_name": "duckyPad",
+    "url": "https://github.com/dekuNukem/",
+    "maintainer": "M4cs",
+    "width": 3,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 17,
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/info.json:9]:
```suggestion
```

File: keyboards/dekunukem/duckypad/keymaps/m4cs/rules.mk [hunk 11]
```diff
@@ -0,0 +1,3 @@
+LTO_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/rules.mk:2]:
```suggestion
```

File: keyboards/dekunukem/duckypad/rules.mk [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+COMMAND_ENABLE = no
+SLEEP_LED_ENABLE = no
+NKRO_ENABLE = no
+RGBLIGHT_ENABLE = no
+
+# TODO: Can we get this working with SPI?
+WS2812_DRIVER = bitbang
+RGB_MATRIX_ENABLE = WS2812
+
+OLED_DRIVER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/rules.mk:17]:
```suggestion
# MCU name
MCU = STM32F072

BOARD = ST_STM32F072B_DISCOVERY

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output

# TODO: Can we get this working with SPI?
RGB_MATRIX_ENABLE = WS2812
WS2812_DRIVER = bitbang

OLED_DRIVER_ENABLE = yes
```

File: keyboards/dekunukem/duckypad/readme.md [hunk 13]
```diff
@@ -0,0 +1,6 @@
+# ![duckyPad](https://github.com/dekuNukem/duckyPad)
+
+A 15 Key Macropad w/ RGB and OLED
+
+* Keyboard Maintainer: [M4cs](https://github.com/M4cs) and [half-cambodian-hacker-man](https://github.com/half-cambodian-hacker-man)
+* Hardware Supported: All duckyPad Revisions
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/readme.md:6]:
```suggestion
* Hardware Supported: All duckyPad Revisions

Make example for this keyboard (after setting up your build environment):

    make dekunukem/duckypad:default

Flashing example for this keyboard:

    make dekunukem/duckypad:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

```

**Review by fauxpark — CHANGES_REQUESTED** [review]
This board is missing a default keymap.

**Comment by M4cs** [comment]
@fauxpark I've made changes to resolve your comments. Thank you :)

File: keyboards/dekunukem/duckypad/config.h [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+#define I2C1_SCL 8
+#define I2C1_SDA_BANK GPIOB
+#define I2C1_SDA 9
+
+#define I2C1_SCL_PAL_MODE 1
+#define I2C1_SDA_PAL_MODE 1
+
+#define I2C1_OPMODE OPMODE_I2C
+#define I2C1_CLOCK_SPEED 400000
+#define I2C1_DUTY_CYCLE FAST_DUTY_CYCLE_2
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/config.h:54]: These values are not valid for I2Cv2, if needing to reconfigure you will require <https://docs.qmk.fm/#/i2c_driver?id=i2cv2>
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/dekunukem/duckypad/bootloader_defs.h [hunk 15]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/bootloader_defs.h:1]: This file should be removed

File: keyboards/dekunukem/duckypad/rules.mk [hunk 16]
```diff
@@ -0,0 +1,24 @@
+# MCU Name
+MCU = STM32F072
+
+BOARD = ST_STM32F072B_DISCOVERY
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/rules.mk:4]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/dekunukem/duckypad/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+
+enum layer_codes {
+  RGB_LAYER = SAFE_RANGE
+};
+
+enum my_keycodes {
+    Z0 = SAFE_RANGE, Z1, Z2, \
+    Z3, Z4, Z5, \
+    Z6, Z7, Z8, \
+    Z9, Z10, Z11, \
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/default/keymap.c:39]:
```suggestion
    Z0 = SAFE_RANGE, Z1, Z2,
    Z3, Z4, Z5,
    Z6, Z7, Z8,
    Z9, Z10, Z11,
```

File: keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+enum layer_codes {
+  RGB_LAYER = SAFE_RANGE,
+  ALT_LAYER
+};
+
+enum my_keycodes {
+    Z0 = SAFE_RANGE, Z1, Z2, \
+    Z3, Z4, Z5, \
+    Z6, Z7, Z8, \
+    Z9, Z10, Z11, \
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/keymap.c:41]:
```suggestion
    Z0 = SAFE_RANGE, Z1, Z2,
    Z3, Z4, Z5,
    Z6, Z7, Z8,
    Z9, Z10, Z11,
```

File: keyboards/dekunukem/duckypad/readme.md [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+
+* Keyboard Maintainer: [M4cs](https://github.com/M4cs) and [half-cambodian-hacker-man](https://github.com/half-cambodian-hacker-man)
+
+* Hardware Supported: All duckyPad Revisions
+
+Make example for this keyboard (after setting up your build environment):
+
+```bash
+make dekunukem/duckypad:default
+```
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/readme.md:13]:
```suggestion
    make dekunukem/duckypad:default
```

File: keyboards/dekunukem/duckypad/readme.md [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+
+Flashing example for this keyboard:
+
+```bash
+make dekunukem/duckypad:default:flash
+```
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/readme.md:19]:
```suggestion
    make dekunukem/duckypad:default:flash
```

File: keyboards/dekunukem/duckypad/rules.mk [hunk 21]
```diff
@@ -0,0 +1,22 @@
+# MCU Name
+MCU = STM32F072
+
+# Build Options
+# 	change yes to no to disable
+#	change no to yes to enable
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/rules.mk:6]:
```suggestion
```

File: keyboards/dekunukem/duckypad/rules.mk [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+BOOTMAGIC_ENABLE = no		# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no		# Mouse Functionality
+EXTRAKEY_ENABLE = yes		# Audio Control and System Controls
+CONSOLE_ENABLE = no			# Console for Debugging
+COMMAND_ENABLE = no			# Commands for Debugging and Configuration
+SLEEP_LED_ENABLE = no		# Breating Sleep LED during USB Suspend
+NKRO_ENABLE = no 			# USB NKey Rollover | If this doesn't work, see here:  https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = no		# Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/rules.mk:17]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by M4cs** [comment]
@fauxpark Thanks for the suggested changes. I've gone ahead and fixed up my files.

File: keyboards/dekunukem/duckypad/readme.md [hunk 23]
```diff
@@ -0,0 +1,20 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+
+    make dekunukem/duckypad:default
+
+
+Flashing example for this keyboard:
+
+    make dekunukem/duckypad:default:flash
+
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/readme.md:18]:
```suggestion
    make dekunukem/duckypad:default

Flashing example for this keyboard:

    make dekunukem/duckypad:default:flash
```

File: keyboards/dekunukem/duckypad/rules.mk [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+MOUSEKEY_ENABLE = no        # Mouse Functionality
+EXTRAKEY_ENABLE = yes       # Audio Control and System Controls
+CONSOLE_ENABLE = no         # Console for Debugging
+COMMAND_ENABLE = no         # Commands for Debugging and Configuration
+SLEEP_LED_ENABLE = no       # Breating Sleep LED during USB Suspend
+# If this doesn't work, see here:  https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB NKey Rollover
+RGBLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/rules.mk:17]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/dekunukem/duckypad/readme.md [hunk 25]
```diff
@@ -0,0 +1,20 @@
+# ![duckyPad](https://github.com/dekuNukem/duckyPad)
+
+A 15 Key Macropad w/ RGB and OLED
+
+* Keyboard Maintainer: [M4cs](https://github.com/M4cs) and [half-cambodian-hacker-man](https://github.com/half-cambodian-hacker-man)
+
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/readme.md:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dekunukem/duckypad/rules.mk [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = no         # Console for Debugging
+COMMAND_ENABLE = no         # Commands for Debugging and Configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breating Sleep LED during USB Suspend
+# If this doesn't work, see here:  https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB NKey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/dekunukem/duckypad/rules.mk:19]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by M4cs** [comment]
Is my build failing due to other keyboards or is it mine?

**Comment by fauxpark** [comment]
The former, but you can rebase this branch onto current master to make Travis build only this board.

**Review by fauxpark — APPROVED** [review]

**Comment by M4cs** [comment]
> The former, but you can rebase this branch onto current master to make Travis build only this board.

Thanks! Merged qmk upstream master into my branch, should fix the issue :)

File: keyboards/dekunukem/duckypad/duckypad.c [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+}, {
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+} };
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A0);
+    writePinHigh(A0);
+    writePinLow(A0);
+    wait_ms(10);
+    writePinHigh(A0);
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/duckypad.c:50]:
```suggestion
    writePinHigh(A0);
    
    keyboard_pre_init_user();
```

File: keyboards/dekunukem/duckypad/keymaps/default/config.h [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/keymaps/default/config.h:20]:
```suggestion
#pragma once

```

File: keyboards/dekunukem/duckypad/keymaps/m4cs/config.h [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
```
Comment (zvecr) [inline keyboards/dekunukem/duckypad/keymaps/m4cs/config.h:20]:
```suggestion
#pragma once

```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by M4cs** [comment]
Fixed thanks @zvecr +1

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**